output file:
processed_korniasample_is_valid_for_homography240.json
function:
sample_is_valid_for_homography
Error Cases:

Pass or Failed: 0

Related Failed Test Cases:
{'../publishablew/kornia/kornia/tests/geometry/test_homography.py::TestSampleValidation::test_bad[cpu-float32] FAILED', 'FAILED ../publishablew/kornia/kornia/tests/geometry/test_homography.py::TestSampleValidation::test_batch[cpu-float32]', 'FAILED ../publishablew/kornia/kornia/tests/geometry/test_homography.py::TestSampleValidation::test_bad[cpu-float32]', '../publishablew/kornia/kornia/tests/geometry/test_homography.py::TestSampleValidation::test_batch[cpu-float32] FAILED', 'FAILED ../publishablew/kornia/kornia/tests/geometry/test_homography.py::TestSampleValidation::test_good[cpu-float32]', '../publishablew/kornia/kornia/tests/geometry/test_homography.py::TestSampleValidation::test_good[cpu-float32] FAILED'}

All Test Cases On Generated code:
Setting up torch compile...
============================= test session starts ==============================
platform linux -- Python 3.10.12, pytest-8.3.3, pluggy-1.5.0 -- /local/data0/moved_data/publishablew/kornia/kornia/venv/bin/python
cachedir: .pytest_cache

cpu info:
	- Model name: AMD Ryzen 7 PRO 5845 8-Core Processor
	- Architecture: x86_64
	- CPU(s): 16
	- Thread(s) per core: 2
	- CPU max MHz: 4661.7178
	- CPU min MHz: 2200.0000
gpu info: {'GPU 0': 'NVIDIA GeForce RTX 3060'}
main deps:
    - kornia-0.7.4
    - torch-2.5.1+cu124
        - commit: a8d6afb511a69687bbb2b7e88a3cf67917e1697e
        - cuda: 12.4
        - nvidia-driver: 555.42.02
x deps:
    - accelerate-1.1.1
dev deps:
    - kornia_rs-0.1.7
    - onnx-1.17.0
gcc info: (Ubuntu 10.5.0-1ubuntu1~22.04) 10.5.0
available optimizers: {'', 'openxla', 'inductor', 'jit', 'cudagraphs', 'tvm', 'onnxrt', None}
model weights cached: ['checkpoints']

rootdir: /local/data0/moved_data/publishablew/kornia/kornia
configfile: pyproject.toml
plugins: timeout-2.3.1
collecting ... collected 3 items

../publishablew/kornia/kornia/tests/geometry/test_homography.py::TestSampleValidation::test_good[cpu-float32] FAILED
../publishablew/kornia/kornia/tests/geometry/test_homography.py::TestSampleValidation::test_bad[cpu-float32] FAILED
../publishablew/kornia/kornia/tests/geometry/test_homography.py::TestSampleValidation::test_batch[cpu-float32] FAILED

=================================== FAILURES ===================================
_________________ TestSampleValidation.test_good[cpu-float32] __________________

self = <tests.geometry.test_homography.TestSampleValidation object at 0x7df4b89060b0>
device = device(type='cpu'), dtype = torch.float32

    def test_good(self, device, dtype):
        pts1 = torch.tensor([[0.0, 0.0], [0.0, 1.0], [1.0, 1.0], [1.0, 0.0]], device=device, dtype=dtype)[None]
        mask = sample_is_valid_for_homography(pts1, pts1)
        expected = torch.tensor([True], device=device, dtype=torch.bool)
>       assert torch.equal(mask, expected)
E       assert False
E        +  where False = <built-in method equal of type object at 0x7df58d6bf1c0>(tensor([[[False, False, False],\n         [False, False, False],\n         [False, False, False]]]), tensor([True]))
E        +    where <built-in method equal of type object at 0x7df58d6bf1c0> = torch.equal

../publishablew/kornia/kornia/tests/geometry/test_homography.py:28: AssertionError
__________________ TestSampleValidation.test_bad[cpu-float32] __________________

self = <tests.geometry.test_homography.TestSampleValidation object at 0x7df4b89064a0>
device = device(type='cpu'), dtype = torch.float32

    def test_bad(self, device, dtype):
        pts1 = torch.tensor([[0.0, 0.0], [1.0, 0.0], [0.0, 1.0], [1.0, 1.0]], device=device, dtype=dtype)[None]
    
        pts2 = torch.tensor([[0.0, 0.0], [0.0, 1.0], [1.0, 0.0], [1.0, 1.0]], device=device, dtype=dtype)[None]
        mask = sample_is_valid_for_homography(pts1, pts2)
        expected = torch.tensor([False], device=device, dtype=torch.bool)
>       assert torch.equal(mask, expected)
E       assert False
E        +  where False = <built-in method equal of type object at 0x7df58d6bf1c0>(tensor([[[False, False, False],\n         [False, False, False],\n         [False, False, False]]]), tensor([False]))
E        +    where <built-in method equal of type object at 0x7df58d6bf1c0> = torch.equal

../publishablew/kornia/kornia/tests/geometry/test_homography.py:36: AssertionError
_________________ TestSampleValidation.test_batch[cpu-float32] _________________

self = <tests.geometry.test_homography.TestSampleValidation object at 0x7df4b89067d0>
device = device(type='cpu'), dtype = torch.float32

    def test_batch(self, device, dtype):
        batch_size = 5
        pts1 = torch.rand(batch_size, 4, 2, device=device, dtype=dtype)
        pts2 = torch.rand(batch_size, 4, 2, device=device, dtype=dtype)
        mask = sample_is_valid_for_homography(pts1, pts2)
>       assert mask.shape == torch.Size([batch_size])
E       assert torch.Size([5, 3, 3]) == torch.Size([5])
E         
E         Left contains 2 more items, first extra item: 3
E         
E         Full diff:
E         - torch.Size([5])
E         + torch.Size([5, 3, 3])
E         ?              ++++++

../publishablew/kornia/kornia/tests/geometry/test_homography.py:43: AssertionError
=========================== short test summary info ============================
FAILED ../publishablew/kornia/kornia/tests/geometry/test_homography.py::TestSampleValidation::test_good[cpu-float32]
FAILED ../publishablew/kornia/kornia/tests/geometry/test_homography.py::TestSampleValidation::test_bad[cpu-float32]
FAILED ../publishablew/kornia/kornia/tests/geometry/test_homography.py::TestSampleValidation::test_batch[cpu-float32]
============================== 3 failed in 0.33s ===============================


Final Test Result:
Setting up torch compile...
============================= test session starts ==============================
platform linux -- Python 3.10.12, pytest-8.3.3, pluggy-1.5.0 -- /local/data0/moved_data/publishablew/kornia/kornia/venv/bin/python
cachedir: .pytest_cache

cpu info:
	- Model name: AMD Ryzen 7 PRO 5845 8-Core Processor
	- Architecture: x86_64
	- CPU(s): 16
	- Thread(s) per core: 2
	- CPU max MHz: 4661.7178
	- CPU min MHz: 2200.0000
gpu info: {'GPU 0': 'NVIDIA GeForce RTX 3060'}
main deps:
    - kornia-0.7.4
    - torch-2.5.1+cu124
        - commit: a8d6afb511a69687bbb2b7e88a3cf67917e1697e
        - cuda: 12.4
        - nvidia-driver: 555.42.02
x deps:
    - accelerate-1.1.1
dev deps:
    - kornia_rs-0.1.7
    - onnx-1.17.0
gcc info: (Ubuntu 10.5.0-1ubuntu1~22.04) 10.5.0
available optimizers: {'', 'onnxrt', 'openxla', 'jit', 'cudagraphs', 'inductor', None, 'tvm'}
model weights cached: ['checkpoints']

rootdir: /local/data0/moved_data/publishablew/kornia/kornia
configfile: pyproject.toml
plugins: timeout-2.3.1
collecting ... collected 3 items

../publishablew/kornia/kornia/tests/geometry/test_homography.py::TestSampleValidation::test_good[cpu-float32] PASSED
../publishablew/kornia/kornia/tests/geometry/test_homography.py::TestSampleValidation::test_bad[cpu-float32] PASSED
../publishablew/kornia/kornia/tests/geometry/test_homography.py::TestSampleValidation::test_batch[cpu-float32] PASSED

=============================== warnings summary ===============================
tests/geometry/test_homography.py::TestSampleValidation::test_good[cpu-float32]
  /local/data0/moved_data/publishablew/kornia/kornia/kornia/geometry/homography.py:239: UserWarning: Using torch.cross without specifying the dim arg is deprecated.
  Please either pass the dim explicitly or simply use torch.linalg.cross.
  The default value of dim will change to agree with that of linalg.cross in a future release. (Triggered internally at ../aten/src/ATen/native/Cross.cpp:62.)
    torch.cross(src_perm[..., 1:2, :], src_perm[..., 2:3, :]) @ src_perm[..., 0:1, :].permute(0, 1, 3, 2)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================= 3 passed, 1 warning in 0.16s =========================


Initial Result:
Setting up torch compile...
============================= test session starts ==============================
platform linux -- Python 3.10.12, pytest-8.3.3, pluggy-1.5.0 -- /local/data0/moved_data/publishablew/kornia/kornia/venv/bin/python
cachedir: .pytest_cache

cpu info:
	- Model name: AMD Ryzen 7 PRO 5845 8-Core Processor
	- Architecture: x86_64
	- CPU(s): 16
	- Thread(s) per core: 2
	- CPU max MHz: 4661.7178
	- CPU min MHz: 2200.0000
gpu info: {'GPU 0': 'NVIDIA GeForce RTX 3060'}
main deps:
    - kornia-0.7.4
    - torch-2.5.1+cu124
        - commit: a8d6afb511a69687bbb2b7e88a3cf67917e1697e
        - cuda: 12.4
        - nvidia-driver: 555.42.02
x deps:
    - accelerate-1.1.1
dev deps:
    - kornia_rs-0.1.7
    - onnx-1.17.0
gcc info: (Ubuntu 10.5.0-1ubuntu1~22.04) 10.5.0
available optimizers: {'', 'onnxrt', None, 'cudagraphs', 'inductor', 'openxla', 'tvm', 'jit'}
model weights cached: ['checkpoints']

rootdir: /local/data0/moved_data/publishablew/kornia/kornia
configfile: pyproject.toml
plugins: timeout-2.3.1
collecting ... collected 3 items

../publishablew/kornia/kornia/tests/geometry/test_homography.py::TestSampleValidation::test_good[cpu-float32] PASSED
../publishablew/kornia/kornia/tests/geometry/test_homography.py::TestSampleValidation::test_bad[cpu-float32] PASSED
../publishablew/kornia/kornia/tests/geometry/test_homography.py::TestSampleValidation::test_batch[cpu-float32] PASSED

=============================== warnings summary ===============================
tests/geometry/test_homography.py::TestSampleValidation::test_good[cpu-float32]
  /local/data0/moved_data/publishablew/kornia/kornia/kornia/geometry/homography.py:239: UserWarning: Using torch.cross without specifying the dim arg is deprecated.
  Please either pass the dim explicitly or simply use torch.linalg.cross.
  The default value of dim will change to agree with that of linalg.cross in a future release. (Triggered internally at ../aten/src/ATen/native/Cross.cpp:62.)
    torch.cross(src_perm[..., 1:2, :], src_perm[..., 2:3, :]) @ src_perm[..., 0:1, :].permute(0, 1, 3, 2)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================= 3 passed, 1 warning in 0.17s =========================
